src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -37,15 +38,24 @@ public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE,
                 PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_MODULE);
 
-        if (!areAllPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL)
+        if (!(areAllPrefixesPresent(argMultimap, PREFIX_NAME)
+                && areAnyPrefixesPresent(argMultimap, PREFIX_PHONE, PREFIX_EMAIL))
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+
+        Phone phone = null;
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        }
+
+        Email email = null;
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        }
 
         Address address = null;
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
```

Comment:
Actually not sure if there is a better way parse the attributes (I feel like this may be consider as bad use of Optional) but since the parse method here throws the ParseException, it cannot be wrapped in a lamda function, unless we put a try catch block inside the lamda function, which is even worse. 

I also tried to write a Functional Interface and a method that returns a function to wrapped up this part, 
```
   public static <T, R> Function<T,R> wrapParser(ParserFunction<T,R> parser) {
        return value -> {
            try {
                return parser.apply(value);
            } catch (ParseException e) {
                throw new RuntimeException(e);
            }
        };
    }

    @FunctionalInterface
    public interface ParserFunction<T, R> {
        R apply(T t) throws ParseException;
    }
```

It works, however, maybe due to I catch it and then throw a ``RuntimeException`` there, the behaviour is not as expected when there is ``ParseException``.